furnace/pid1.py

```diff
@@ -28,18 +28,25 @@
 from pathlib import Path
 
 from furnace.libc import unshare, mount, umount2, non_caching_getpid, pivot_root, is_mount_point, \
-    MS_BIND, MS_REC, MS_SLAVE, CLONE_NEWPID, CLONE_NEWNET, MNT_DETACH
-from furnace.config import NAMESPACES, CONTAINER_MOUNTS, CONTAINER_DEVICE_NODES, HOSTNAME
+    MS_BIND, MS_REC, MS_SLAVE, MS_REMOUNT, MS_RDONLY, CLONE_NEWPID, CLONE_NEWNET, MNT_DETACH
+from furnace.config import NAMESPACES, CONTAINER_MOUNTS, CONTAINER_DEVICE_NODES, HOSTNAME, BindMount
 
 logger = logging.getLogger("container.pid1")
 
 
 class PID1:
-    def __init__(self, root_dir: Path, control_read, control_write, isolate_networking):
+    def __init__(self, root_dir: Path, control_read, control_write, isolate_networking, bind_mounts):
         self.control_read = control_read
         self.control_write = control_write
         self.root_dir = root_dir.resolve()
         self.isolate_networking = isolate_networking
+        self.bind_mounts = []
+        for source, destination, read_only in bind_mounts:
+            source = Path(source)
+            destination = Path(destination)
+            if destination.is_absolute():
+                destination = destination.relative_to("/")
+            self.bind_mounts.append(BindMount(source, destination, read_only))
```

Comment:
A separate method would be nice for making this list. E.g. `_make_bind_mounts` or something.
so this would be `self.bind_mounts = self._make_bind_mounts(bind_mounts)`